server/src/main/java/org/elasticsearch/health/node/DiskHealthIndicatorService.java

```diff
@@ -0,0 +1,448 @@
+    }
+
+    private String getSymptom(
+        boolean clusterHasBlockedIndex,
+        Set<String> blockedIndices,
+        Set<String> nodesWithBlockedIndices,
+        Set<String> nodesReportingRed,
+        Set<String> nodesReportingYellow,
+        ClusterState clusterState
+    ) {
+        Set<String> allUnhealthyNodes = (Stream.concat(
+            Stream.concat(nodesWithBlockedIndices.stream(), nodesReportingRed.stream()),
+            nodesReportingYellow.stream()
+        )).collect(Collectors.toSet());
+        Set<String> allRolesOnUnhealthyNodes = getRolesOnNodes(allUnhealthyNodes, clusterState).stream()
+            .map(DiscoveryNodeRole::roleName)
+            .collect(Collectors.toSet());
+        final String symptom;
+        if (clusterHasBlockedIndex && allUnhealthyNodes.isEmpty()) {
+            // In this case the disk issue has been resolved but the index block has not been automatically removed yet:
+            symptom = String.format(
+                Locale.ROOT,
+                "%d %s blocked and cannot be updated but 0 nodes are currently out of space.",
+                blockedIndices.size(),
+                blockedIndices.size() == 1 ? "index is" : "indices are"
+            );
+        } else {
+            symptom = String.format(
+                Locale.ROOT,
+                "%d node%s with roles: [%s] %s out of disk or running low on disk space.",
+                allUnhealthyNodes.size(),
+                allUnhealthyNodes.size() == 1 ? "" : "s",
+                allRolesOnUnhealthyNodes.stream().sorted().collect(Collectors.joining(", ")),
+                allUnhealthyNodes.size() == 1 ? "is" : "are"
+            );
+        }
+        return symptom;
+    }
+
+    private List<HealthIndicatorImpact> getImpacts(
+        Set<String> indicesWithBlock,
+        Set<String> indicesOnRedNodes,
+        Set<String> indicesOnYellowNodes,
+        Set<String> nodesWithBlockedIndices,
+        Set<String> redDataNodes,
+        Set<String> yellowDataNodes,
+        Set<String> redMasterNodes,
+        Set<String> yellowMasterNodes,
+        Set<String> redNonDataNonMasterNodes,
+        Set<String> yellowNonDataNonMasterNodes
+    ) {
+        List<HealthIndicatorImpact> impacts = new ArrayList<>();
+        if (indicesWithBlock.isEmpty() == false
+            || indicesOnRedNodes.isEmpty() == false
+            || nodesWithBlockedIndices.isEmpty() == false
+            || redDataNodes.isEmpty() == false) {
+            impacts.add(
+                new HealthIndicatorImpact(1, "Cannot insert or update documents in the affected indices.", List.of(ImpactArea.INGEST))
+            );
+        } else if (indicesOnYellowNodes.isEmpty() == false || yellowDataNodes.isEmpty() == false) {
+            impacts.add(
+                new HealthIndicatorImpact(
+                    1,
+                    "At risk of not being able to insert or update documents in the affected indices.",
```

Comment:
I think this sentence is a bit confusing.

It is missing a subject. What is at risk? 

Let's update this in a subsequent PR (create an issue for this)

Reply:
I will revisit this as part of https://github.com/elastic/elasticsearch/issues/90212